core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -679,10 +700,23 @@ private TableMetadata setCurrentSnapshotTo(Snapshot snapshot) {
 
   public TableMetadata replaceProperties(Map<String, String> newProperties) {
     ValidationException.check(newProperties != null, "Cannot set properties to null");
-    return new TableMetadata(null, formatVersion, uuid, location,
+    Map<String, String> validProperties = newProperties;
+    int newFormatVersion = formatVersion;
+    if (newProperties.containsKey(PROPERTY_FORMAT_VERSION)) {
+      validProperties = Maps.newHashMap(newProperties);
+      newFormatVersion = Integer.parseInt(validProperties.remove(PROPERTY_FORMAT_VERSION));
+    }
```

Comment:
I see lots of this code block in this PR, I'd like to have a small abstracted static method:

```java
  private static int parseFormatVersion(Map<String, String> properties, int defaultFormatVersion) {
    return properties.containsKey(PROPERTY_FORMAT_VERSION) ?
        Integer.parseInt(properties.remove(PROPERTY_FORMAT_VERSION)) : defaultFormatVersion;
  }
```

Then we could simplify this block as: 

```java
    Map<String, String> validProperties = Maps.newHashMap(newProperties);
    int newFormatVersion = parseFormatVersion(validProperties, formatVersion);
```

Reply:
We also already have `PropertyUtil` that you can use to make this one line.